File: 2356-largest-combination-with-bitwise-and-greater-than-zero/largest-combination-with-bitwise-and-greater-than-zero.go
func largestCombination(candidates []int) int {
    res := 0
    maxC := int(slices.Max(candidates))

    for bit := 1; bit <= maxC; bit = bit<<1 {
        count := 0
        for _, c := range candidates {
            if c & bit > 0 {
                count++
            }
        }
        if count > res {
            res = count
        }
    }
    return res
}

File: 2356-largest-combination-with-bitwise-and-greater-than-zero/largest-combination-with-bitwise-and-greater-than-zero.c
int largestCombination(int* candidates, int candidatesSize) {
    int maxSize = 0;
    for (int i = 0; i < 24; i++) {
        int count = 0;
        for (int j = 0; j < candidatesSize; j++) {
            if ((candidates[j] & (1 << i)) != 0) {
                count++;
            }
        }
        if (count > maxSize) {
            maxSize = count;
        }
    }
    return maxSize;
}